pandas/core/generic.py

```diff
@@ -3636,6 +3637,34 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
         Returns
         -------
         casted : type of caller
+
+        Examples
```

Comment:
For me it works as well with numpy dtypes:

```
In [77]: s1 = pd.Series([1,2])

In [78]: s2 = s1.astype('int', copy=False)

In [79]: s2[0] = 10

In [80]: s1
Out[80]: 
0    10
1     2
dtype: int64
```

It's just that the dtype needs to be equivalent (otherwise it always takes a copy).

So I would change the example